Wikipedia/Code/LoadingAnimationViewController.swift

```diff
@@ -0,0 +1,90 @@
+
+import UIKit
+
+class LoadingAnimationViewController: UIViewController {
+    
+    var theme: Theme = Theme.standard
+    
+    var cancelBlock: (() -> Void)?
+    @IBOutlet private var satelliteImageView: UIImageView!
+    @IBOutlet private var backgroundView: UIView!
+    @IBOutlet private var cancelButton: UIButton!
+    @IBOutlet private var backgroundImageView: UIImageView!
+    @IBOutlet private var loadingPageLabel: UILabel!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        apply(theme: theme)
+        
+        cancelButton.setTitle(CommonStrings.cancelActionTitle, for: .normal)
+        loadingPageLabel.text = WMFLocalizedString("link-loading-title", value: "Loading page...", comment: "Title displayed in loading overlay after link is tapped.")
+        view.accessibilityViewIsModal = true
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        satelliteImageView.startRotating()
+        UIAccessibility.post(notification: UIAccessibility.Notification.screenChanged, argument: loadingPageLabel)
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        
+        satelliteImageView.stopRotating()
+        UIAccessibility.post(notification: UIAccessibility.Notification.screenChanged, argument: nil)
+    }
+    
+
+    @IBAction func tappedCancel(_ sender: UIButton) {
+        cancelBlock?()
+    }
+    
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        cancelButton.titleLabel?.font = UIFont.wmf_font(.semiboldTitle3, compatibleWithTraitCollection: traitCollection)
+        loadingPageLabel.font = UIFont.wmf_font(.boldHeadline, compatibleWithTraitCollection: traitCollection)
```

Comment:
The fonts should also be set in `viewDidLoad` in case `traitCollectionDidChange` doesn't get called on iOS 13 initially (https://useyourloaf.com/blog/predicting-size-classes-in-ios-13/)

Reply:
@nambatee Done